.github/workflows/linters.yml

```diff
@@ -0,0 +1,32 @@
+name: Format code using black and isort
+# modified from https://towardsdatascience.com/black-with-git-hub-actions-4ffc5c61b5fe
+# caching https://stackoverflow.com/questions/59127258/how-can-i-use-pip-cache-in-github-actions
+# black integration https://black.readthedocs.io/en/stable/integrations/github_actions.html
+# isort integration https://pycqa.github.io/isort/docs/configuration/github_action.html
+
+
+
+on:
+  push:
+    branches: [ "develop", "issue23-precommithooks" ]
+
+jobs:
+  linters:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Format code with black 
+        run: |
+          pip install black==21.12b0
```

Comment:
I worry about this hard-coding of versions in multiple places. Let's research a shell command for extracting the first match to a regex in black, e.g. this regex 'black[^\"]+' should match `black==21.12b`
Same for isort and click. Worst case scenario, we could dynamically create a python script that echos the pip install command.

Reply:
Yes.  That makes sense.  

I have a todo item to work out a way to extract the versions of these tools from pyproject.toml as the source of ground truth.

Reply:
export black_version=$(python -c 'import tomli; print([line for line in  tomli.load(open("pyproject.toml","rb"))["project"]["optional-dependencies"]["dev"] if "black" in line][0])')
 
 is ugly but it puts 'black==21.12b' into the environment var $black_version

Reply:
that's perfect!

Reply:
implemented and checked in